podpac/core/cache/cache.py

```diff
@@ -140,19 +141,20 @@ def has(self, node, key, coordinates=None, mode=None):
 class CacheStore(object):
 
     def get_hash_val(self, obj):
-        return hash(obj)
+        return hash_alg(obj).hexdigest()
 
     def hash_node(self, node):
-        hashable_repr = cPickle.dumps(node.definition)
-        return self.get_hash_val(hashable_repr)
+        hashable_repr = 'None' if node is None else node.hash
+        return hashable_repr 
 
     def hash_coordinates(self, coordinates):
-        hashable_repr = None if coordinates is None else coordinates.json
-        return self.get_hash_val(hashable_repr)
+        hashable_repr = 'None' if coordinates is None else coordinates.hash
+        return hashable_repr 
 
     def hash_key(self, key):
-        hashable_repr = str(repr(key))
-        return self.get_hash_val(hashable_repr)
+        #hashable_repr = str(repr(key)).encode('utf-8')
+        #return self.get_hash_val(hashable_repr)
+        return key
```

Comment:
I didn't hash the key-- felt it was useful to be human-readable string on the filesystem. It also helped me debug something...

Reply:
ok, thats fine, but I'll have to override `hash_key()` in `DiskCacheStore` so that the key is cleansed (e.g. remove file path separators, periods, and the special strings 'kKeY' etc.).

basically run this function on it:

https://github.com/creare-com/podpac/blob/19554a37c20e31a669881ad53d1d67ed43427e40/podpac/core/cache/cache.py#L344

Reply:
Ah! Really need that function elsewhere... I forgot to flag it...